src/rebar_app_discover.erl

```diff
@@ -56,6 +57,27 @@ merge_deps(AppInfo, State) ->
 
     {AppInfo1, State1}.
 
+project_app_config(AppInfo, State) ->
+    C = rebar_config:consult(rebar_app_info:dir(AppInfo)),
+    Dir = rebar_app_info:dir(AppInfo),
+    maybe_reset_hooks(C, Dir, State).
+
+%% Here we check if the app is at the root of the project.
+%% If it is, then drop the hooks from the config so they aren't run twice
+maybe_reset_hooks(C, Dir, State) ->
+    case filename:dirname(rebar_dir:root_dir(State)) of
+        Dir ->
+            C1 = proplists:delete(provider_hooks, C),
+            proplists:delete(hooks, C1);
+        _ ->
+            C
+    end.
+
+reset_hooks(State) ->
+    lists:foldl(fun(Key, StateAcc) ->
+                        rebar_state:set(StateAcc, Key, [])
+                end, State, [post_hooks, pre_hooks, provider_hooks]).
+
```

Comment:
Na, we always want to reset the hooks here. There is not "dir" for the state either to even compare against. The reason deleting them from config is different is because that is then used to build up a state record based on the current state. 

So if the current state isn't wiped of hooks all apps would get them, but in the case of the config we only want to wipe if it is a top level app. This way for a case like `apps/app1` the state of `app1` will not get the `./rebar.config` hooks and it won't wipe its `apps/app1/rebar.config` hooks.